bigquery_etl/cli/backfill.py

```diff
@@ -401,12 +402,27 @@ def initiate(
     )
 
     # deploy backfill staging table
-    ctx.invoke(
-        deploy,
-        name=f"{dataset}.{table}",
-        project_id=project,
-        destination_table=backfill_staging_qualified_table_name,
-    )
+    try:
+        ctx.invoke(
+            deploy,
+            name=f"{project}.{dataset}.{table}",
+            project_id=project,
+            destination_table=backfill_staging_qualified_table_name,
+        )
+    except Exception as e:
+        raise Exception(
+            f"Deploying backfill staging table resulted in error for {qualified_table_name}."
+        ) from e
+        sys.exit(1)
+    else:
+        client = bigquery.Client(project=project)
+
+        # check that backfill staging table exists
+        if not _table_exists(client, backfill_staging_qualified_table_name):
+            raise NotFound(
+                f"Backfill staging table does not exist: {backfill_staging_qualified_table_name}."
+            )
+            sys.exit(1)
```

Comment:
nit: no need to `sys.exit` after raising, the code won't be reachable anyway since raising stops execution